tap_salesforce/salesforce/__init__.py

```diff
@@ -513,3 +523,100 @@ def get_blacklisted_fields(self):
             raise TapSalesforceException(
                 "api_type should be REST or BULK was: {}".format(
                     self.api_type))
+
+    def date_columns_in_filter(self, filter):
+        date_cols = set()
+        for f in filter["filters"]:
+            if f["rhs"]["litType"] == "date":
+                date_cols.add(f["lhs"]["name"])
+
+        return date_cols
+
+    def get_date_column_types(self, date_cols_in_filter):
+        response = self.describe()
+
+        date_col_types = dict()
+
+        if self.source_type == 'object':
+            fields = response['fields']
+            for f in fields:
+                if f['name'] in date_cols_in_filter:
+                    date_col_types[f['name']] = f['type']
+        elif self.source_type == 'report':
+            fields = response['reportExtendedMetadata']['detailColumnInfo']
+            for f in fields.values():
+                if f['label'] in date_cols_in_filter:
+                    date_col_types[f['label']] = f['dataType']
+
+        return date_col_types
```

Comment:
@ChrisLing1 I was doing some testing, and apparently Salesforce likes to really distinguish between date and datetime columns. We consider both just `date`, but if you add the time part to just a date column, the query throws an error. So I added in this extra step to get the salesforce column type if a date column is used in the filter - for SF datetime columns, the time part will be added, but for SF date columns, the time part won't be added. Tested with filters including both date and datetime columns, and it worked.